test/integration.nix

```diff
@@ -1,58 +1,89 @@
+              isNormalUser = true;
+              passwordFile = config.age.secrets.passwordfile-user1.path;
+            };
+          };
+        };
+      };
 
-    age.secrets.passwordfile-user1 = {
-      file = ../example/passwordfile-user1.age;
-    };
+      nodes.system2 = {
+        imports = [
+          ../modules/age.nix
+          ./install_ssh_host_keys.nix
+        ];
 
-    users = {
-      mutableUsers = false;
+        services.openssh = sshdConf;
 
-      users = {
-        user1 = {
-          isNormalUser = true;
-          passwordFile = config.age.secrets.passwordfile-user1.path;
+        age.secrets.ex1 = {
+          file = ../example/passwordfile-user1.age;
+          action = "echo bar > /tmp/foo";
         };
       };
-    };
-
-  };
-
-  testScript =
-  let
-    user = "user1";
-    password = "password1234";
-  in ''
-    system1.wait_for_unit("multi-user.target")
-    system1.wait_until_succeeds("pgrep -f 'agetty.*tty1'")
-    system1.sleep(2)
-    system1.send_key("alt-f2")
-    system1.wait_until_succeeds(f"[ $(fgconsole) = 2 ]")
-    system1.wait_for_unit(f"getty@tty2.service")
-    system1.wait_until_succeeds(f"pgrep -f 'agetty.*tty2'")
-    system1.wait_until_tty_matches(2, "login: ")
-    system1.send_chars("${user}\n")
-    system1.wait_until_tty_matches(2, "login: ${user}")
-    system1.wait_until_succeeds("pgrep login")
-    system1.sleep(2)
-    system1.send_chars("${password}\n")
-    system1.send_chars("whoami > /tmp/1\n")
-    system1.wait_for_file("/tmp/1")
-    assert "${user}" in system1.succeed("cat /tmp/1")
-  '';
-}) args
+
+      nodes.system2After = recursiveUpdate nodes.system2 {
+        age.secrets.ex1.file = ../example/secret1.age;
+      };
+
+      testScript =
+        let
+          user = "user1";
+          password = "password1234";
+        in
+        { nodes, ... }:
+        ''
+          system1.start()
+          system2.start()
+
+          system1.wait_for_unit("multi-user.target")
+          system1.wait_until_succeeds("pgrep -f 'agetty.*tty1'")
+          system1.sleep(2)
+          system1.send_key("alt-f2")
+          system1.wait_until_succeeds("[ $(fgconsole) = 2 ]")
+          system1.wait_for_unit("getty@tty2.service")
+          system1.wait_until_succeeds("pgrep -f 'agetty.*tty2'")
+          system1.wait_until_tty_matches(2, "login: ")
+          system1.send_chars("${user}\n")
+          system1.wait_until_tty_matches(2, "login: ${user}")
+          system1.wait_until_succeeds("pgrep login")
+          system1.sleep(2)
+          system1.send_chars("${password}\n")
+          system1.send_chars("whoami > /tmp/1\n")
+          system1.wait_for_file("/tmp/1")
+          assert "${user}" in system1.succeed("cat /tmp/1")
+
+          system2.wait_for_unit("multi-user.target")
+          system2.wait_until_succeeds("${nodes.system2After.config.system.build.toplevel}/bin/switch-to-configuration test")
+          system2.wait_until_succeeds("grep bar /tmp/foo")
```

Comment:
Seems like we should also test that this *doesn't* get run at startup. i.e. that `/tmp/foo` doesn't exist until we redeploy.